### PR TITLE
Generate separate node certs/keys for client and server authentication

### PR DESCRIPTION
**Pull request by mberhault**
It's better to have separate certificates for server auth and client auth.
Each has their own private key.

Also fixed up some x509 fields and expanded the tests a bit to reflect client cert enforcement.

### CODE REVIEW

File: security/certs.go [hunk 1]
```diff
@@ -28,6 +28,28 @@ import (
 	"github.com/cockroachdb/cockroach/util"
 )
 
+// loadCACertAndKey loads the certificate and key files in the specified
+// directory, parses them, and returns the x509 certificate and private key.
+func loadCACertAndKey(certsDir string) (*x509.Certificate, crypto.PrivateKey, error) {
```
Comment (tamird) [inline security/certs.go]: bit of a nit: why does this return a pointer?

**Comment by tamird** [comment]
LGMT

File: security/certs.go (hunk 1 above)
Comment (MBerhault) [inline security/certs.go]: it's an interface: http://golang.org/pkg/crypto/#PrivateKey

Comment (tamird) [inline security/certs.go]: I was referring to http://golang.org/pkg/crypto/x509/#Certificate

Comment (MBerhault) [inline security/certs.go]: oh. I was reading "why does it not return a pointer".
This is ultimately what x509.CreateCertificate expects. I figured I might as well return a pointer from the get go.

Comment (tamird) [inline security/certs.go]: thought so =/

File: resource/test_certs/README.md [hunk 2]
```diff
@@ -4,8 +4,10 @@ Contains the following files:
 
 * ca.crt: CA certificate
 * ca.key: CA private key
-* node.crt: server certificate
-* node.key: server private key
+* node.crt: node client certificate
```
Comment (bdarnell) [inline resource/test_certs/README.md]: The client certs should probably be named `node.client.*` to be explicit.

**Comment by bdarnell** [comment]
LGTM

File: resource/test_certs/README.md (hunk 2 above)
Comment (MBerhault) [inline resource/test_certs/README.md]: I'm using the "Common Name" for now as I'm still trying to figure out how I'm going to go from Context.User to filename. I probably does make sense to always have *.client.*, but I'll leave that for when I start loading something other than node.crt

Comment (MBerhault) [inline resource/test_certs/README.md]: actually, you're right, it's much much cleaner this way. Done.

**Pull request merged**